### PR TITLE
Extract datadog.yaml file

### PR DESCRIPTION
**Pull request by nullobject**
Rather than copy the example file and edit multiple lines, it's much easier to define our own datadog.yaml file with exactly what we need.

### CODE REVIEW

**Comment by nullobject** [comment]
[Extract datadog.yaml file](https://trello.com/c/oeV6D4kz/11030-extract-datadogyaml-file)

File: datadog.rb [hunk 1]
```diff
@@ -47,22 +55,18 @@ def nginx_dest
...
   met? do
-    "/etc/datadog-agent/datadog.yaml".p.grep(/^api_key: #{datadog_api_key}/) &&
-    "/etc/datadog-agent/datadog.yaml".p.grep(/^use_dogstatsd: yes/) &&
-    "/etc/datadog-agent/datadog.yaml".p.grep(/^dogstatsd_port: 8126/) &&
+    up_to_date?(datadog_src, datadog_dest) &&
     nginx_dest.exists?
   end
 
   meet do
-    sudo %(
```
Comment (nullobject) [inline datadog.rb]: We were substituting lines in the example file, it's much simpler to just have our own config from the start.

File: datadog.rb [hunk 2]
```diff
@@ -38,31 +38,55 @@
   provides "datadog-agent"
 end
 
-dep "datadog configured", :datadog_api_key do
+dep "datadog configured", :datadog_api_key, :datadog_postgres_password do
+  def datadog_src
+    "datadog/datadog.yaml.erb".p
+  end
+
+  def datadog_dest
```
Comment (eleanorakh) [inline datadog.rb]: is dest 'destination'?

**Comment by nickbrowne** [comment]
Nice improvement :+1:

**Pull request merged**